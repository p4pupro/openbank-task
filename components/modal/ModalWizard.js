import React, { useState } from "react";
import { useForm, useStep } from "react-hooks-helper";
import { Info } from "../info/Info";
import { CreatePassword } from "../create/CreatePassword";
import { Feedback } from "../feedback/Feedback";

const steps = [
  { id: "productInfo" },
  { id: "createPassword" },
  { id: "feedback" },
];

const defaultData = {
  password: "",
  confirmPassword: "",
  rememberPassword: "",
};
export const ModalWizard = ({ onRequestClose }) => {
  const [formData, setForm] = useForm(defaultData);
  const { step, navigation } = useStep({ initialStep: 0, steps });
  const { id } = step;
  const [response, setResponse] = useState(401);

  const props = {
    formData,
    setForm,
    navigation,
    onRequestClose,
    response,
    setResponse,
  };

  switch (id) {
    case "productInfo":
      return <Info {...props} />;
    case "createPassword":
      return <CreatePassword {...props} />;
    case "feedback":
      return <Feedback {...props} />;
    default:
      return null;
  }
};
