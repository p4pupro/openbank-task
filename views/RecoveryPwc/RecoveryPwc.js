import React, { useState } from "react";
import OpenbankLogo from "../../assets/img/key_openbank.png";
import LogoOpenBank from "../../assets/img/logo_openbank.png";
import { ModalWizard } from "../../components/modal/ModalWizard";
import { useTranslation } from "react-i18next";

export const RecoveryPwc = () => {
  const [isModalOpen, setModalIsOpen] = useState(false);
  const { t, i18n } = useTranslation();

  const toggleModal = () => {
    setModalIsOpen(!isModalOpen);
  };

  const changeLanguage = () => {
    if (i18n.language === "en") {
      i18n.changeLanguage("es");
    } else {
      i18n.changeLanguage("en");
    }
  };

  return isModalOpen ? (
    <ModalWizard onRequestClose={toggleModal} />
  ) : (
    <div className="container is-fluid">
      <div>
        <img
          src={OpenbankLogo}
          className="App-header-logo"
          alt={"openbank-logo"}
        />
        <h1>
          {t("WELCOME")}
          <strong>{t("MANAGER-PWC")}</strong>
        </h1>

        <a className="" onClick={() => changeLanguage()}>
          <span className="icon lang-color">
            <i className="fas fa-flag"></i>
            {i18n.language === "en" ? <p>ES</p> : <p>EN</p>}
          </span>
        </a>
      </div>

      <hr />
      <section className="section ">
        <div className="container">
          <h1 className="title">{t("MANAGER-PWC")}</h1>
          <h2 className="subtitle">
            {t("SUBTITLE")}{" "}
            <strong>
              <a onClick={toggleModal}>{t("CLICK-HERE")}</a>
            </strong>
            {t("START-PROCCESS")}
          </h2>
        </div>
      </section>
      <footer className="footer">
        <div className="content has-text-centered">
          <p>
            <strong>Openbank task</strong> by{" "}
            <a href="https://github.com/p4pupro">Domingo PÃ©rez</a>
          </p>
          <img
            src={LogoOpenBank}
            className="App-header-logo"
            alt={"openbank-logo"}
          />
        </div>
      </footer>
    </div>
  );
};
