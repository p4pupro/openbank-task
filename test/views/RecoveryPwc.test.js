import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { ModalWizard } from "../../components/modal/ModalWizard";
import { MemoryRouter, Route } from "react-router-dom";
import { RecoveryPwc } from "../../views/RecoveryPwc/RecoveryPwc";

describe("Test - <RecoveryPwc />", () => {
  let wrapper = shallow(<RecoveryPwc />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<RecoveryPwc />);
  });

  test("Should renders fine", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
