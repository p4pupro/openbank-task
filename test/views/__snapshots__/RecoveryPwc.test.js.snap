// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test - <RecoveryPwc /> Should renders fine 1`] = `
<div
  className="container is-fluid"
>
  <div>
    <img
      alt="openbank-logo"
      className="App-header-logo"
      src="key_openbank.png"
    />
    <h1>
      WELCOME
      <strong>
        MANAGER-PWC
      </strong>
    </h1>
    <a
      className=""
      onClick={[Function]}
    >
      <span
        className="icon lang-color"
      >
        <i
          className="fas fa-flag"
        />
        <p>
          EN
        </p>
      </span>
    </a>
  </div>
  <hr />
  <section
    className="section "
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        MANAGER-PWC
      </h1>
      <h2
        className="subtitle"
      >
        SUBTITLE
         
        <strong>
          <a
            onClick={[Function]}
          >
            CLICK-HERE
          </a>
        </strong>
        START-PROCCESS
      </h2>
    </div>
  </section>
  <footer
    className="footer"
  >
    <div
      className="content has-text-centered"
    >
      <p>
        <strong>
          Openbank task
        </strong>
         by
         
        <a
          href="https://github.com/p4pupro"
        >
          Domingo PÃ©rez
        </a>
      </p>
      <img
        alt="openbank-logo"
        className="App-header-logo"
        src="logo_openbank.png"
      />
    </div>
  </footer>
</div>
`;
